src/System.CommandLine/ChildList.cs

```diff
@@ -37,7 +38,35 @@ public ChildList(Command parent)
         public void Add(T item)
         {
             item.AddParent(_parent);
+
+            if (_children.Count > 0 && _parent is RootCommand)
```

Comment:
`RootCommand` ctor adds `HelpOption` and `VersionOption` by default now. Users might want to customize it like that:

```cs
RootCommand rootCommand = new()
{
    new VersionOption("-v", "-version"),
};
```

In order to avoid duplicates, this method checks if a help option or version option already exists. If it does, it overrides it.

Reply:
I think it might be less surprising to just ask people to remove the other instance, especially given that the `HelpAction` is  also customizable and is probably the more common path to customizing help.

Reply:
@jonsequitur you are right, I've removed this logic and updated the tests